### ISSUE TITLE
si-icon: text descriptions for accessibility

### ISSUE THREAD

**Issue by michael-smt** [siemens/element#1890]
### Prerequisites

- [x] I have read the [Contributing Guidelines](https://github.com/siemens/element/blob/main/CONTRIBUTING.md).
- [x] I have not leaked any internal/restricted information like screenshots, videos, code snippets, links etc.

### Feature request

`<si-icon>` should provide a default textual description of the icon which can also be overridden by the application.

### Use case

See e.g. [simpl/simpl-element#2696](https://code.siemens.com/simpl/simpl-element/-/work_items/2696), specifically the thread in [comment 53813651](https://code.siemens.com/simpl/simpl-element/-/work_items/2696#note_53813651)

**Comment by spliffone** [comment]
The description has an effect the bundle-size and doesn't work with the icon-font. Additionally based on the context the icon can have a different meaning. Within the library we have plenty of places where the icon shouldn't have a descriptive text since e.g. it is only representative or the parent already provide the description.

Can I suggest to discuss the request in the core-team-meeting?

**Comment by michael-smt** [comment]
Thanks for the feedback @spliffone. Yes let's discuss it.

Maybe the scope should be more narrow: not about tooltips but only for a `alt` and/or `aria-label` on the icon that describes the visual of the icon. Like for example "fork and knife" for the `element-dinner` icon or "three water drops" for `element-humidity`.

**Comment by michael-smt** [comment]
Might be related to
- #1902

**Comment by Danisand** [comment]
> The description has an effect the bundle-size and doesn't work with the icon-font. Additionally based on the context the icon can have a different meaning. Within the library we have plenty of places where the icon shouldn't have a descriptive text since e.g. it is only representative or the parent already provide the description.
> 
> Can I suggest to discuss the request in the core-team-meeting?

@spliffone Yes, it might be that a single icon is used for several equipment types. But the equipment type should be unique within the SI B ecosystem. Therefore I suggest to invert the approach:

1. Select the the semantic (e.g. `Air handling unit`)
2. Use the icon mapping based on semantic (see [bmx/bmx-semantics#45](https://code.siemens.com/bmx/bmx-semantics/-/work_items/45))

@michael-smt The translated text of the semantic could be used for the ARIA labels. Or do I see something wrong?

**Comment by michael-smt** [comment]
In my view with `<si-icon>` you just get a icon. And that icon's ARIA label should only describes how the icon looks. So for example `element-welcome` would say "two hands interlocked in a handshake", or `element-room-segment` described as "rectangle with rounded edges with a vertical dotted line in the middle".

A semantic description should then be provided by a parent element, preferably as a Buildings component that wraps `<si-icon>`.

Or maybe I'm totally off track with my ideas 😉

**Comment by Danisand** [comment]
@michael-smt The ARIA-label should show the function in my point of view (e.g., `Campus` is much more helpful than `Multiple buildings`)

The visual description could be the last fallback for the ARIA-label if the usage is not known.

**Comment by spike-rabbit** [comment]
`si-icon` used to support this but we decided to remove this on purpose:
- in most cases (at least in Element) we actually don't want icons to have description, they are aria-hidden on purpose, so a default would require many overrides
- we cannot provide a meaningful default, `element-trash` would probably be `Delete` in most cases unless you want to symbolize a trash can.
- arrows are even harder to default because in RTL one would need to reverse the direction

The recommendations for apps is use the axe scanner and make sure everything needed is labelled.
